vendor/engines/secure_rooms/app/models/secure_rooms/card_reader.rb

```diff
@@ -4,6 +4,9 @@ class CardReader < ActiveRecord::Base
 
     belongs_to :secure_room, foreign_key: :product_id
 
+    validates :card_reader_number, :control_device_number, presence: true
+    validates :card_reader_number, uniqueness: true
```

Comment:
That might be a question for @danhodos. I thought reader_number would be something like 1, 2, or 3. I thought the controller number was more like a serial.

This validation should match the index, though, which is on both fields.